tox.ini

```diff
@@ -1,19 +1,60 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/pure-python
 [tox]
 envlist =
-    py27,pypy,py34,py35,py36,py37,pypy3,coverage
+    lint,
+    py27,
+    pypy,
+    py35,
+    py36,
+    py37,
+    py38,
+    pypy3,
+    coverage
 
 [testenv]
+usedevelop = true
+deps =
+    zope.testrunner
 commands =
     zope-testrunner --test-path=src []
-deps =
-    .[test]
+extras = test
 
+[testenv:lint]
+basepython = python3
+skip_install = true
+deps =
+    flake8
+    check-manifest
+    check-python-versions
+commands =
+    flake8 src setup.py
+    check-manifest
+    check-python-versions .
 
 [testenv:coverage]
-usedevelop = true
-commands =
-    coverage run -m zope.testrunner --test-path=src []
-    coverage report --fail-under=100
+basepython = python3
 deps =
-    {[testenv]deps}
     coverage
+    coverage-python-version
+    zope.testrunner
+commands =
+    coverage run -m zope.testrunner --test-path=src []
+    coverage html
+    coverage report -m
+
+[coverage:run]
+branch = True
+plugins = coverage_python_version
+source = src
+
+[coverage:report]
+precision = 2
+exclude_lines =
+    pragma: nocover
+    except ImportError:
+    raise NotImplementedError
+    if __name__ == '__main__':
```

Comment:
FWIW the old .coveragerc also had the following in exclude_lines:

```
    self.fail
    raise AssertionError
```

We might want to look into that, if we notice a coverage drop.

(The GitHub review UI is bad in some respects: I can't see commit statuses while I'm writing review comments, unless I open new tabs and poke around.)

Reply:
Coveralls did not leave a commit status, so I guess we're not tracking coverage for this package.

(Or coveralls once again ran into GitHub API rate limits and can't post commit statuses.)

Reply:
I can't even log in to coveralls.io due to a GitHub-Error in the log-in process.

So I am going to add the missing ignores blindly.